app/src/test/java/com/arygm/quickfix/model/search/AnnouncementRepositoryFirestoreTest.kt

```diff
@@ -0,0 +1,364 @@
+    verify(mockAnnouncementDocument).set(eq(announcement1))
+  }
+
+  @Test
+  fun updateAnnouncement_onSuccess_callsOnSuccess() {
+    whenever(mockCollectionReference.document(announcement1.announcementId))
+        .thenReturn(mockDocumentReference)
+    whenever(mockDocumentReference.set(announcement1)).thenReturn(Tasks.forResult(null))
+
+    var callbackCalled = false
+
+    announcementRepositoryFirestore.updateAnnouncement(
+        announcement = announcement1,
+        onSuccess = { callbackCalled = true },
+        onFailure = { fail("Failure callback should not be called") })
+    shadowOf(Looper.getMainLooper()).idle()
+    assert(callbackCalled)
+  }
+
+  @Test
+  fun updateAnnouncement_onFailure_callsOnFailure() {
+    val exception = Exception("Test exception")
+    whenever(mockCollectionReference.document(announcement1.announcementId))
+        .thenReturn(mockDocumentReference)
+    whenever(mockDocumentReference.set(announcement1)).thenReturn(Tasks.forException(exception))
+
+    var failureCallbackCalled = false
+
+    announcementRepositoryFirestore.updateAnnouncement(
+        announcement = announcement1,
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { e ->
+          failureCallbackCalled = true
+          assertEquals(exception, e)
+        })
+    shadowOf(Looper.getMainLooper()).idle()
+    assert(failureCallbackCalled)
+  }
+
+  @Test
+  fun deleteAnnouncementById_onSuccess_callsOnSuccess() {
+    val userId = "user1"
+    val announcementId = "id1"
+    whenever(mockCollectionReference.document(announcementId)).thenReturn(mockDocumentReference)
+    whenever(mockDocumentReference.delete()).thenReturn(Tasks.forResult(null))
+
+    var callbackCalled = false
+
+    announcementRepositoryFirestore.deleteAnnouncementById(
+        userId = userId,
+        announcementId = announcementId,
+        onSuccess = { callbackCalled = true },
+        onFailure = { fail("Failure callback should not be called") })
+    shadowOf(Looper.getMainLooper()).idle()
+    assert(callbackCalled)
+  }
+
+  @Test
+  fun deleteAnnouncementById_onFailure_callsOnFailure() {
+    val userId = "user1"
+    val announcementId = "id1"
+    val exception = Exception("Test exception")
+    whenever(mockCollectionReference.document(announcementId)).thenReturn(mockDocumentReference)
+    whenever(mockDocumentReference.delete()).thenReturn(Tasks.forException(exception))
+
+    var failureCallbackCalled = false
+
+    announcementRepositoryFirestore.deleteAnnouncementById(
+        userId = userId,
+        announcementId = announcementId,
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { e ->
+          failureCallbackCalled = true
+          assertEquals(exception, e)
+        })
+    shadowOf(Looper.getMainLooper()).idle()
+    assert(failureCallbackCalled)
+  }
+
+  private fun invokeDocumentToAnnouncement(document: DocumentSnapshot): Announcement? {
```

Comment:
Good job fir this one !! I will use this method to test private method in my backend features also. Thanks (: !!